/jds/bin/diff -uprN gnome-utils-2.26.0.old/baobab/src/baobab-utils.c gnome-utils-2.26.0/baobab/src/baobab-utils.c
--- gnome-utils-2.26.0.old/baobab/src/baobab-utils.c	2009-03-19 15:18:55.422186000 +0000
+++ gnome-utils-2.26.0/baobab/src/baobab-utils.c	2009-03-19 15:24:04.344024000 +0000
@@ -43,36 +43,14 @@
 void
 baobab_get_filesystem (baobab_fs *fs)
 {
-	size_t i;
-	glibtop_mountlist mountlist;
-	glibtop_mountentry *mountentries;
+    glibtop_fsusage fsusage;
 
+    /* OpenSolaris using ZFS filesystem , just need to get slash usage stats */
+    glibtop_get_fsusage (&fsusage, "/");
 	memset (fs, 0, sizeof *fs);
-
-	mountentries = glibtop_get_mountlist (&mountlist, FALSE);
-
-	for (i = 0; i < mountlist.number; ++i) {
-		GFile	*file;
-		glibtop_fsusage fsusage;
-		
-		file = g_file_new_for_path(mountentries[i].mountdir);	
-
-		if (baobab_is_excluded_location (file)){
-			g_object_unref(file);
-			continue;
-			}
-
-		glibtop_get_fsusage (&fsusage, mountentries[i].mountdir);
-
-		/*  v.1.1.1 changed bavail with bfree) */
-		fs->total += fsusage.blocks * fsusage.block_size;
-		fs->avail += fsusage.bfree * fsusage.block_size;
-		fs->used += (fsusage.blocks - fsusage.bfree) * fsusage.block_size;
-		g_object_unref(file);
-
-	}
-
-	g_free (mountentries);
+	fs->total = fsusage.blocks * fsusage.block_size;
+	fs->avail = fsusage.bfree * fsusage.block_size;
+	fs->used = (fsusage.blocks - fsusage.bfree) * fsusage.block_size;
 }
 
 void
